AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp

```diff
@@ -0,0 +1,163 @@
+/** @file AdvancedLoggerMmCoreGoogleTest.cpp
+
+    This file contains the unit tests for the Advanced Logger MM Core Library.
+
+    Copyright (c) Microsoft Corporation.
+    SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+#include <GoogleTest/Library/MockHobLib.h>
+#include <GoogleTest/Library/MockAdvancedLoggerHdwPortLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Library/BaseLib.h>
+  #include <Library/DebugLib.h>
+  #include <AdvancedLoggerInternal.h>
+  #include <Protocol/AdvancedLogger.h>
+  #include <AdvancedLoggerInternalProtocol.h>
+  #include <Library/SynchronizationLib.h>       // to mock (MU_BASECORE MdePkg)
+  #include "../../AdvancedLoggerCommon.h"
+
+  extern ADVANCED_LOGGER_INFO  *mLoggerInfo;
+  extern UINT32                mBufferSize;
+  extern EFI_PHYSICAL_ADDRESS  mMaxAddress;
+  extern BOOLEAN               mInitialized;
+
+  // Static function declaration
+  BOOLEAN
+  ValidateInfoBlock (
+    VOID
+    );
+}
+
+using namespace testing;
+
+/**
+  Test class for AdvancedLoggerMmCore
+**/
+class AdvancedLoggerMmCoreTest : public Test {
+protected:
+  UINTN DebugLevel;
+  EFI_HANDLE ImageHandle;
+  EFI_SYSTEM_TABLE SystemTable;
+  BOOLEAN Status;
+  ADVANCED_LOGGER_INFO testLoggerInfo;
+  // StrictMock<MockHobLib> gHobLib;
+  // StrictMock<MockAdvancedLoggerHdwPortLib> gALHdwPortLib;
+
+  void
+  SetUp (
+    ) override
+  {
+    mLoggerInfo                     = NULL;
+    DebugLevel                      = DEBUG_ERROR;
+    ImageHandle                     = (EFI_HANDLE)0x12345678;
+    testLoggerInfo.Signature        = ADVANCED_LOGGER_SIGNATURE;
+    testLoggerInfo.Version          = ADVANCED_LOGGER_VERSION;
+    testLoggerInfo.LogBufferOffset  = (ALIGN_VALUE (sizeof (testLoggerInfo), 8));
+    testLoggerInfo.LogCurrentOffset = (ALIGN_VALUE (sizeof (testLoggerInfo), 8));
+
+  }
+};
+
+//
+// Test ValidateInfoBlock
+//
+TEST_F (AdvancedLoggerMmCoreTest, AdvLoggerGetInfoFail) {
+  // NULL LoggerInfo
+  Status = ValidateInfoBlock ();
+  EXPECT_EQ (Status, FALSE);
+
+  // Invalid Signature
+  mLoggerInfo            = &testLoggerInfo;
+  mLoggerInfo->Signature = SIGNATURE_32 ('T', 'E', 'S', 'T');
+  Status                 = ValidateInfoBlock ();
+  EXPECT_EQ (Status, FALSE);
+  mLoggerInfo->Signature = ADVANCED_LOGGER_SIGNATURE;
+
+  // Mismatched Version is okay? Wouldn't expect mismatched version with valid signature?

```

Comment:
We should check Version, too. The signature tells us we have the right structure (well, hopefully), the version may be diferent if this MMCore instance is sharing a log buffer with DXE that thinks it is different, etc.

Reply:
Made change in https://github.com/microsoft/mu_plus/pull/582 to add a version check, will update the test.